wallet/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1593,6 +1597,54 @@ instance BuildableSafeGen WalletSoftwareUpdate where
         updBlockchainVersion
         updScriptVersion
 
+data WalletBalance = WalletBalance
+  { wbBalance  :: V1 (Core.Coin)
+  , wbWalletId :: Text
+  } deriving (Show, Eq, Generic)
+deriveJSON Aeson.defaultOptions ''WalletBalance
+
+instance ToSchema WalletBalance where
+    declareNamedSchema =
+        genericSchemaDroppingPrefix "wb" (\(--^) props -> props
+            & "balance"
+            --^ "the balance of a given public key"
+        )
+
+instance Example WalletBalance
+instance Arbitrary WalletBalance where
+    arbitrary = WalletBalance <$> arbitrary <*> arbitrary
+
+instance Buildable WalletBalance where
+    build (WalletBalance bal walid) = "WalletBalance { ballence = " <> (show bal) <> ", walletid = " <> show walid <> " }"
+
+instance ToSchema EncryptedSecretKey where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "EncryptedSecretKey") $ mempty
+            & type_ .~ SwaggerString
+            & format ?~ "hex|base16"
+
+instance ToJSON EncryptedSecretKey where
+  toJSON _esk = "todo"
```

Comment:
This is dubious. Why would you serialize an ESK to JSON ?

Reply:
the servant/swagger stuff refuses to compile without that type
i'm not sure if i should just hard-code it to returning a constant value, or actually encoding it as hex

Reply:
I'd recommend to use an invariant here `error "toJSON was called on EncryptedSecretKey"`